src/interface/variable_pack.hpp

```diff
@@ -0,0 +1,234 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//=======================================================================================
+#ifndef INTERFACE_VARIABLE_PACK_HPP_
+#define INTERFACE_VARIABLE_PACK_HPP_
+
+#include <array>
+#include <forward_list>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "interface/metadata.hpp"
+#include "interface/variable.hpp"
+
+namespace parthenon {
+
+// some convenience aliases
+namespace vpack_types {
+template <typename T>
+using VarList = std::forward_list<std::shared_ptr<CellVariable<T>>>;
+using IndexPair = std::pair<int, int>;
+using StringPair = std::pair<std::vector<std::string>, std::vector<std::string>>;
```

Comment:
I think these need some explanation. What is do the first and second index of the `IndexPair` represent? What is the purpose for the `StringPair`?

Reply:
`IndexPair` is explained in the user documentation. This is what ends up in the `PackIndexMap`. These represent the inclusive  bounds of the indices that represent a variable in the variable pack.

`StringPair` is the key of the cache for fluxpacks, because you need the names of the variables and the names of the fluxes, and order matters.

Reply:
I added some additional explanatory comments inline.